### PR TITLE
SDXL: fix gradient accumulation bug and add clip grad in graph mode

### PR DESCRIPTION
**Pull request by Songyuanwei**
1. add clip score
2. fix gradient accumulation bug :  accumulated_grad = accumulated_grad  + grads  ——>  accumulated_grad = accumulated_grad + grads / accum_steps

### CODE REVIEW

**Review by SamitHuang — APPROVED** [review]

File: examples/stable_diffusion_xl/gm/models/trainer_factory.py [hunk 1]
```diff
@@ -62,70 +65,91 @@ def construct(
...
+            loss, self.hyper_map(F.partial(_grad_accum_op, self.accum_steps), self.accumulated_grads, grads)
+        )
+        if self.accum_step % self.accum_steps == 0:
+            if self.clip_grad:
+                grads = ops.clip_by_global_norm(self.accumulated_grads, self.clip_norm)
+                loss = F.depend(loss, self.optimizer(grads))
+            else:
+                loss = F.depend(loss, self.optimizer(self.accumulated_grads))
+            loss = F.depend(loss, self.hyper_map(F.partial(_grad_clear_op), self.accumulated_grads))
+            ops.print("parameters were updated.")
```
Comment (zhtmike) [inline examples/stable_diffusion_xl/gm/models/trainer_factory.py:105]: ops.print may be be necessary and harm the performance

Comment (Songyuanwei) [reply examples/stable_diffusion_xl/gm/models/trainer_factory.py]: delete

**Review by zhanghuiyao — APPROVED** [review]
